mreg_cli/host.py

```diff
@@ -2327,6 +2338,182 @@ def srv_show(args):
+    """
+
+    def _delete_sshfp_record(sshfp: dict, hname: str):
+        path = f"/sshfps/{sshfp['id']}"
+        history.record_delete(path, sshfp, redoable=False)
+        delete(path)
+        cli_info("removed SSHFP record with fingerprint {} for {}".format(sshfp["fingerprint"],
+                                                               hname),
+            print_msg=True)
+
+    # Get host info or raise exception
+    info = host_info_by_name(args.name)
+    hid = info["id"]
+
+    # Get all matching SSHFP records
+    path = f"/sshfps/?host={hid}"
+    history.record_get(path)
+    sshfps = get_list(path)
+    if len(sshfps) < 1:
+        cli_warning("no SSHFP records matching {}".format(info["name"]))
+
+    if args.fingerprint:
+        found = False
+        for sshfp in sshfps:
+            if sshfp["fingerprint"] == args.fingerprint:
+                _delete_sshfp_record(sshfp, info["name"])
+                found = True
+        if not found:
+            cli_info("found no SSHFP record with fingerprint {} for {}".format(args.fingerprint,
+                                                               info["name"]),
+                 print_msg=True)
+    else:
+        for sshfp in sshfps:
+            _delete_sshfp_record(sshfp, info["name"])
+            
+
+# Add 'sshfp_remove' as a sub command to the 'host' command
+host.add_command(
+    prog='sshfp_remove',
+    description='Remove SSHFP record with a given fingerprint from the host. '
+                'A missing fingerprint removes all SSHFP records for the host.',
+    callback=sshfp_remove,
+    flags=[
+        Flag('name',
+             description='Host target name.',
+             metavar='NAME'),
+        Flag('-fingerprint',
+             description='Hexadecimal fingerprint.',
+             metavar='FINGERPRINT'),
+    ],
+)
+
+
+##############################################
+# Implementation of sub command 'sshfp_show' #
+##############################################
+
+def sshfp_show(args):
+    """Show SSHFP records. An empty input showes all existing SSHFP records
+    """
+
+    if args.name:
+        # Get host info or raise exception
+        info = host_info_by_name(args.name)
+        hid = info["id"]
+
+        # Get all matching SSHFP records
+        path = f"/sshfps/?host={hid}"
+        history.record_get(path)
+        sshfps = get_list(path)
+        if len(sshfps) < 1:
+            cli_warning("no SSHFP records matching {}".format(info["name"]))
+    else:
+        # Get all SSHFPs
+        path = f"/sshfps/"
+        history.record_get(path)
+        sshfps = get_list(path)
+        if len(sshfps) < 1:
+            cli_warning("SSHFP records are not found")
+    
+    padding = 0
+    prev_name = ""
+    for sshfp in sorted(sshfps, key=lambda k: k["host"]):
+        host_info = host_info_by_id(sshfp["host"])
+        hname = host_info["name"]
+        if len(hname) > padding:
+            padding = len(hname)
```

Comment:
This will cause the padding to get longer and longer and thus look a bit weird. Possible to check names first to get proper padding?

Reply:
Will be gone in the next commit.